db/fetcher/fetcher.go

```diff
@@ -370,12 +523,58 @@ func (df *DocumentFetcher) FetchNext(ctx context.Context) (*encodedDocument, err
 			return nil, err
 		}
 
-		end, err := df.nextKey(ctx)
+		if df.filter != nil {
+			// only run filter if we've collected all the fields
+			// required for filtering. This is tracked by the bitsets.
+			if df.filterSet.Equal(df.doc.filterSet) {
+				df.ranFilter = true
```

Comment:
todo: Minor, this looks like it is set in the wrong location - if errors are produced `ranFilter` will be true, yet the filter may not have been run.  I think this should be set after the error check for `mapper.RunFilter` (currently between ln 539 and 540)

Reply:
Ahh, yea it should be. I don't think it makes a diff since it will error out anyway, but for sanity 👍